### PR TITLE
Fix join mode + part command and other minor fixed issues

### PR DESCRIPTION
**Pull request by faboussard**
After Celia's review, I fixed the following bugs:

**Join:**
No special characters allowed: join #hey’ -> invalid, bad mask.
If I’m already in the channel, I only get a popup message. Ideally, it should send nothing at all.
Issue: When I’ve already joined too many channels and try to rejoin a channel I’m already in, I shouldn’t get this error.
The sendJoinMsg and sendPartMsg functions were removed to avoid broken pipe issues and message spam.

**Part:**
Command added.
An empty channel should be deleted automatically. Could you retest this? I feel like it’s working now.

**Mode:**
If we just input + or -, it now returns.
The uint8 type for the limit caused a value like int 65 to be converted into A.
Entering 0 as the limit now disables the limit.
Overflow is controlled.
I removed the sendNotice functions that were cluttering things.

**Parsing:**
Messages longer than 512 characters are truncated.
Empty messages are ignored (in addition to trimming /r and /n).

I didn’t look into the other authentication errors or the issues related to the ":" that Celia mentioned, as the current functionality seems fine. We’ve also ensured that it works well with both nc and HexChat.

Note:

For privmsg, if the sender sends a message in only one channel, they won’t receive it themselves. I haven’t found another solution, as HexChat secretly adds the #channel even when it’s not explicitly typed. Changing HexChat’s parameters might solve this, but the school’s version doesn’t seem to allow this anyway.

### CODE REVIEW

File: includes/Channel.hpp [hunk 1]
```diff
@@ -74,7 +74,7 @@ class Channel {
   const Mode &getMode(void) const;
   const std::string getChannelModeFlag(void) const;
   const std::string &getKey(void) const;
-  uint8_t getLimit(void) const;
+  int getLimit(void) const;
```
Comment (ysengoku) [inline includes/Channel.hpp:77]: Same as above:
I wonder if it's better to use unsigned type if we won't have negative values. (It'll depend on how you use it in the code.)

**Comment by ysengoku** [comment]
This version seems have an issue in receiving several messages in loop even if they are not sent at once (Each message is less than 512 bytes). We should review message size checking.

**Comment by ysengoku** [comment]
I close this PR without accepting.
Please make another PR without `Parsing class` part changes first, and then should we check this issue  ?

File: includes/Channel.hpp [hunk 2]
```diff
@@ -53,7 +53,7 @@ class Channel {
   Topic _topic;
   Mode _mode;
   std::string _key;
-  uint8_t _limit;
+  int _limit;
```
[the remark this answers is not in the document]
Comment (faboussard) [reply includes/Channel.hpp:56]: it has been changed, it will appear in next PR

**Pull request closed without merging**